opencti-platform/opencti-front/src/private/components/data/DataTableToolBar.jsx

```diff
@@ -721,11 +764,25 @@ class DataTableToolBar extends Component {
     const { t } = this.props;
     const { actionsInputs } = this.state;
     const disabled = R.isNil(actionsInputs[i]?.type) || R.isEmpty(actionsInputs[i]?.type);
-    const typesHaveScore = selectedTypes.every((selectedType) => typesWithScore.includes(selectedType))
-      && entityTypeFilterValues.every((filterType) => typesWithScore.includes(filterType));
-    const typesHaveStatus = selectedTypes.length === 1 // the proposed statuses values depends on the entity type, so we should have only one type selected
-      && !typesWithoutStatus.includes(selectedTypes[0]);
+
+    const checkTypes = (typesList) => selectedTypes.every((type) => typesList.includes(type))
+      && entityTypeFilterValues.every((type) => typesList.includes(type));
+
+    const dynamicOptions = [
```

Comment:
smart :)

Reply:
you could also integrate the options depending only on the action type. This way you construct the final array directly (only need to sort)